Visemes Creation

Yusarina opened the pull request:
**Do not merge this unto the translations are merge again my main branch is kinda messed up due to not have translation in it's own branch.** 

This is not ready for prime time, I still need to add stuff and make this system better however we need to get the armture slelection done before I continue. However the viseme creation itself seems to work mostly fine, does need a little bit of work though. This is very much inspired via the Cats one and some of the code has been used form cats, though some of it has been changed.

To-Do
- Mesh auto detection from selected Armature.
- Auto detection of shapekeys on said mesh.
- Translations.
- Maybe some optimizations on the creation.

Yusarina commented:
Fixed Conflicts

989onan commented:
This needs a revisit, we have translations now so this may need conflicts resolved and integrated with translations

989onan commented on core/common.py:
```diff
@@ -114,4 +114,76 @@ def duplicatebone(b: bpy.types.EditBone) -> bpy.types.EditBone:
\ No newline at end of file
+    return cb
+
+def has_shapekeys(mesh_obj: Object) -> bool:
```
this is instanciated twice, must have done that in the chaos

989onan replied:
other than this, looks good!

989onan reviewed (changes requested):
Reviewed, needs a small change

Yusarina commented:
Oh lol, I already was making changes lol 

- Viesmes will now use selected armature.
- New dropdown menu in the viseme UI so the user can select which mesh to create visemes on.
- New helper function get_armature_meshes
- Added a new check before we create the new visemes to see if any exist, if there do we will remove them and create the new ones.
- fixed several issues and errors.

989onan reviewed (approved):
looks good, perfect!
